cmd/crank/beta/render/render.go

```diff
@@ -150,24 +154,63 @@ func Render(ctx context.Context, in Inputs) (Outputs, error) { //nolint:gocyclo
 	// the desired state returned by the last, and each Function may produce
 	// results.
 	for _, fn := range in.Composition.Spec.Pipeline {
-		req := &fnv1beta1.RunFunctionRequest{Observed: o, Desired: d, Context: fctx}
+		// used to store the resources fetched at the previous iteration
+		var extraResources map[string]*fnv1beta1.Resources
+		// used to store the requirements returned at the previous iteration
+		var requirements *fnv1beta1.Requirements
```

Comment:
Maybe `existing` or `previous` would make sense as a variable name? Could help make the `requirements = req` assignment below more self-documenting.